docs/01-infrastructure-os.md

```diff
@@ -0,0 +1,195 @@
++--------------------------------------+-------------+-----------+------------+--------------------------------------+
+| 110fc25a-6cc7-409f-9b8f-40be05884203 | None        | None      |            | None                                 |
+| 2327d33b-e497-4006-87e3-7991810b1686 | udp         | None      |            | 6f6399ef-b69b-49cb-9f97-8fcad96715bf |
+| 2dfe89ce-c167-4f75-89df-a7bc3007336d | icmp        | 0.0.0.0/0 |            | None                                 |
+| 2e175bd1-f885-41de-97af-0787be7fba9e | tcp         | 0.0.0.0/0 | 3389:3389  | None                                 |
+| 39eaea13-92f5-438b-929c-d7585c84e4b2 | tcp         | 0.0.0.0/0 | 22:22      | None                                 |
+| 5acea256-84b0-420d-923f-f257fe4e7319 | tcp         | 0.0.0.0/0 | 6443:6443  | None                                 |
+| b6bc42d0-9f3f-4dcf-a5b1-7196968320d3 | tcp         | None      |            | 6f6399ef-b69b-49cb-9f97-8fcad96715bf |
+| d1038338-bf4b-4f25-8c29-a104d74c2803 | None        | None      |            | None                                 |
++--------------------------------------+-------------+-----------+------------+--------------------------------------+
+```
+
+## Provision Virtual Machines
+
+All the VMs in this lab will be provisioned using Ubuntu 16.04 mainly because it runs a newish Linux Kernel that has good support for Docker.
+
+### Virtual Machines
+
+#### Kubernetes Controllers
+
+```
+openstack server create --image ubuntu-16.04 --flavor m1.small \
+  --security-group kubernetes --key-name tbritten \
+  --nic net-id=1f9ce4ba-2203-4dc2-b411-c0b35ac588c8,v4-fixed-ip=10.240.0.10 \
+  controller0
+```
+
+```
+openstack server create --image ubuntu-16.04 --flavor m1.small \
+  --security-group kubernetes --key-name tbritten \
+  --nic net-id=1f9ce4ba-2203-4dc2-b411-c0b35ac588c8,v4-fixed-ip=10.240.0.11 \
+  controller1
+```
+
+```
+openstack server create --image ubuntu-16.04 --flavor m1.small \
+  --security-group kubernetes --key-name tbritten \
+  --nic net-id=1f9ce4ba-2203-4dc2-b411-c0b35ac588c8,v4-fixed-ip=10.240.0.12 \
+  controller2
+```
+
+#### Kubernetes Workers
+
+```
+openstack server create --image ubuntu-16.04 --flavor m1.small \
+  --security-group kubernetes --key-name tbritten \
+  --nic net-id=1f9ce4ba-2203-4dc2-b411-c0b35ac588c8,v4-fixed-ip=10.240.0.20 \
+  worker0
+```
+
+```
+openstack server create --image ubuntu-16.04 --flavor m1.small \
+  --security-group kubernetes --key-name tbritten \
+  --nic net-id=1f9ce4ba-2203-4dc2-b411-c0b35ac588c8,v4-fixed-ip=10.240.0.21 \
+  worker1
+```
+
+```
+openstack server create --image ubuntu-16.04 --flavor m1.small \
+  --security-group kubernetes --key-name tbritten \
+  --nic net-id=1f9ce4ba-2203-4dc2-b411-c0b35ac588c8,v4-fixed-ip=10.240.0.22 \
+  worker2
+```
+
+### Kubernetes Public Address
+
+Attached a floating IP to the controller0 to allow for remote access:
```

Comment:
At least in the cloud I tried this on (citycloud) I needed to create the floating ip first:

```
openstack floating ip create ext-net
```